src/sql/workbench/services/objectExplorer/browser/asyncServerTreeRenderer.ts

```diff
@@ -159,16 +190,21 @@ class TreeNodeTemplate extends Disposable {
 	private _icon: HTMLElement;
 	private _labelContainer: HTMLElement;
 	private _label: ResourceLabel;
+	private _actionBar: ActionBar;
 
 	constructor(
 		container: HTMLElement,
-		@IInstantiationService private readonly _instantiationService: IInstantiationService
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IObjectExplorerService private _objectExplorerService: IObjectExplorerService
 	) {
 		super();
 		this._root = dom.append(container, dom.$('.object-element-container'));
 		this._icon = dom.append(this._root, dom.$('div.object-icon'));
 		this._labelContainer = dom.append(this._root, dom.$('div.label'));
 		this._label = this._instantiationService.createInstance(ResourceLabel, this._labelContainer, { supportHighlights: true });
+		const actionsContainer = dom.append(this._label.element.element, dom.$('.actions'));
+		this._actionBar = new ActionBar(actionsContainer, {
```

Comment:
So we have to manually add on the inline actions for the async tree? I would have expected that'd be built into the tree itself, is this because we're doing our own custom rendering of the tree items?

Reply:
Yes, because the actions items are present in treeview renderer